src/main/java/oop/animals/enclosures/EnclosureService.java

```diff
@@ -0,0 +1,42 @@
+package oop.animals.enclosures;
+
+import oop.animals.Animal;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class EnclosureService<T extends Animal> implements EnclosureServiceInterface<T> {
+    private final List<Enclosure<T>> enclosures = new ArrayList<>();
+
+    private void createEnclosure() {
+        Enclosure<T> enclosure = new Enclosure<>();
+        enclosures.add(enclosure);
+    }
+
+    private void createEnclosureAndAddAnimal(T animal) {
+        Enclosure<T> enclosure = new Enclosure<>();
+        enclosures.add(enclosure);
+        enclosure.addAnimal(animal);
+    }
+
+    @Override
+    public void addAnimalToSpecificEnclosure(T animal, Enclosure<T> enclosure) {
+        enclosure.addAnimal(animal);
+    }
+
+    @Override
+    public void addAnimalToAnySuitableEnclosure(T animal) {
+        for (Enclosure<T> enclosure : enclosures) {
+            if (enclosure.getAnimals().size() < 5) {
+                enclosure.addAnimal(animal);
+            } else {
+                createEnclosureAndAddAnimal(animal);
```

Comment:
If you have 4 enclosures in the list with 5 animals, how many new enclosures will be created here ?